File: python/algorithm/sorting/selection.py
def sort(nums):
  for i in range (len(nums) - 1, 0, -1):
    maxValue = nums[0]
    idx = 0
    for j in range(0, i + 1):
      if nums[j] > maxValue:
        maxValue = nums[j]
        idx = j
    nums[idx] = nums[i]
    nums[i] = maxValue
  return nums

File: js/algorithm/sorting/selection.js
module.exports = function (nums) {
  for (let i = nums.length - 1; i > 0; i--) {
    let max = nums[0]
    let idx = 0
    for (let j = 0; j <= i; j++) {
      if (nums[j] > max) {
        max = nums[j]
        idx = j
      }
    }
    nums[idx] = nums[i]
    nums[i] = max
  }
  return nums
}
